pkg/discovery/dtofactory/node_entity_dto_builder.go

```diff
@@ -221,11 +226,108 @@ func (builder *nodeEntityDTOBuilder) getNodeCommoditiesSold(node *api.Node, clus
 		Capacity(clusterCommodityDefaultCapacity).
 		Create()
 	if err != nil {
-		return nil, err
+		return nil, isAvailableForPlacement, err
 	}
 	commoditiesSold = append(commoditiesSold, clusterComm)
 
-	return commoditiesSold, nil
+	return commoditiesSold, isAvailableForPlacement, nil
+}
+
+// getNodeStorageCommoditiesSold builds sold storage commodities for the node
+// Returns the built commodities and if this node is available for placement or not.
+// The availability for placement is evaluated based on the current usage crossing the
+// configured threshold. If the usage has crossed the threshold, we mark the node
+// NOT available for placement.
+func (builder *nodeEntityDTOBuilder) getNodeStorageCommoditiesSold(nodeName string) ([]*proto.CommodityDTO, bool) {
+	var resourceCommoditiesSold []*proto.CommodityDTO
+	vstorageResources := []string{"rootfs", "imagefs"}
```

Comment:
Nit: can you create constants for "rootfs" and "imagefs" to avoid unexpected typo when using in multiple places?